### PR TITLE
Use PosixPath for path handling and create parent folders if they not exist

### PR DESCRIPTION
**Pull request by CM000n**
This should fix an error if the parents of the CACHE_FILENAME do not exist in some environments.
See: https://github.com/DurgNomis-drol/mytoyota/issues/270

### CODE REVIEW

**Comment by CM000n** [comment]
You should now have maintainer access @GitOldGrumpy 😊 Can you pls review this MR?

File: mytoyota/controller.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 _LOGGER: logging.Logger = logging.getLogger(__package__)
 
-CACHE_FILENAME: Optional[str] = expanduser("~/.cache/toyota_credentials_cache_contains_secrets")
+CACHE_FILENAME: Path = Path.home() / ".cache" / "toyota_credentials_cache_contains_secrets"
```
Comment (DurgNomis-drol) [inline mytoyota/controller.py:25]: Why do we store the credentials?

Comment (CM000n) [reply mytoyota/controller.py:25]: See discussion here https://github.com/DurgNomis-drol/mytoyota/pull/242#issuecomment-1843433435 and here https://github.com/DurgNomis-drol/mytoyota/pull/242#issuecomment-1843750651

**Review by GitOldGrumpy — APPROVED** [review]
Looks good to me.

**Comment by CM000n** [comment]
> Looks good to me.

Strange, merging ist still blocked :-/
Have you received and accepted the Maintainer invitation? At the moment it looks like you are still not registered as a repo Maintainer.

**Review by GitOldGrumpy — APPROVED** [review]

**Comment by GitOldGrumpy** [comment]
> > Looks good to me.
> 
> Strange, merging ist still blocked :-/ Can you please approve these changes @GitOldGrumpy?
> 
> Have you received and accepted the Maintainer invitation E-Mail? You are still shown here as Contributor instead of Caollaborator.

Yes I had seen the email, didn't realise I had to accept! I've done that now and it approved. Know cant merge because of open conversations.

**Pull request merged**